cmd/skaffold/app/cmd/fix.go

```diff
@@ -51,39 +51,47 @@ func doFix(_ context.Context, out io.Writer) error {
 }
 
 func fix(out io.Writer, configFile string, toVersion string, overwrite bool) error {
-	cfg, err := schema.ParseConfig(configFile)
+	parsedCfgs, err := schema.ParseConfig(configFile)
 	if err != nil {
 		return err
 	}
-
-	if cfg.GetVersion() == toVersion {
+	needsUpdate := false
+	for _, cfg := range parsedCfgs {
+		if cfg.GetVersion() != toVersion {
+			needsUpdate = true
+			break
+		}
+	}
+	if !needsUpdate {
 		color.Default.Fprintln(out, "config is already version", toVersion)
 		return nil
 	}
 
-	cfg, err = schema.ParseConfigAndUpgrade(configFile, toVersion)
+	versionedCfgs, err := schema.ParseConfigAndUpgrade(configFile, toVersion)
 	if err != nil {
 		return err
 	}
 
 	// TODO(dgageot): We should be able run validations on any schema version
 	// but that's not the case. They can only run on the latest version for now.
 	if toVersion == latest.Version {
-		if err := validation.Process(cfg.(*latest.SkaffoldConfig)); err != nil {
+		var cfgs []*latest.SkaffoldConfig
+		for _, cfg := range versionedCfgs {
+			cfgs = append(cfgs, cfg.(*latest.SkaffoldConfig))
+		}
+		if err := validation.Process(cfgs); err != nil {
 			return fmt.Errorf("validating upgraded config: %w", err)
 		}
 	}
-
-	newCfg, err := yaml.Marshal(cfg)
+	newCfg, err := yaml.MarshalWithSeparator(versionedCfgs)
 	if err != nil {
 		return fmt.Errorf("marshaling new config: %w", err)
 	}
-
 	if overwrite {
 		if err := ioutil.WriteFile(configFile, newCfg, 0644); err != nil {
 			return fmt.Errorf("writing config file: %w", err)
 		}
-		color.Default.Fprintf(out, "New config at version %s generated and written to %s\n", cfg.GetVersion(), opts.ConfigurationFile)
+		color.Default.Fprintf(out, "New config at version %s generated and written to %s\n", latest.Version, opts.ConfigurationFile)
```

Comment:
> was `cfg.GetVersion()` always `latest.Version`?

it should be `toVersion` instead of `latest.Version`. 

> this seems like it's always going to combine multiple configs into one single file with separators

This function is called per file so it will upgrade all configs in the same file and write them back out the same file. When we modify `opts.ConfigurationFile` to be a list of file paths, we'll modify this function accordingly.